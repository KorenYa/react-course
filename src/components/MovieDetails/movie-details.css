.movie-details {
    padding: 60px 20px;
    background: url('https://assets.nflxext.com/ffe/siteui/vlv3/a27b8244-8c0b-4db4-984e-0cd57daa489d/8e38e13b-8cfc-410e-8d2d-a6c9e717d9b8/UA-en-20190506-popsignuptwoweeks-perspective_alpha_website_large.jpg') center center;
    color: #ffffff;
}

.movie-details__title {
    font-size: 18px;
    font-weight: 600;
    color: #f55263;
    padding-bottom: 10px;
    text-transform: uppercase;
}

.movie-details__tagline {
    font-size: 14px;
    font-weight: 400;
    padding: 10px 0;
}

.movie-details__date, .movie-details__runtime {
    font-weight: 600;
    padding-right: 15px;
}

.movie-details__overview {
    font-size: 14px;
    font-weight: 400;
    padding-top: 20px;
}