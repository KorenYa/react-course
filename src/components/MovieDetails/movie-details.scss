@import '../../styles/constants.scss';

.movie-details {
    padding: 60px 20px;
    background: url('../../../src/img/bg.jpg') center center;
    color: $white;

    &__title {
        font-size: $lg;
        font-weight: $bold;
        color: $red;
        padding-bottom: 10px;
        text-transform: uppercase;
    }

    &__tagline {
        font-size: $md;
        font-weight: $thin;
        padding: 10px 0;
    }

    &__date,
    &__runtime {
        font-weight: $bold;
        padding-right: 15px;
    }

    &__overview {
        font-size: $md;
        font-weight: $thin;
        padding-top: 20px;
    }
}
