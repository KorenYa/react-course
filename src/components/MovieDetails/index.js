import React from 'react';
import Image from '../Image';
import PropTypes from 'prop-types';
import './movie-details.scss';
const MovieDetails = function(props) {
    const movie = props.movie;
    return (
        <div className='movie-details'>
            <div className='row'>
                <div className='col-4'>
                    <Image src={movie.poster_path} alt={movie.title} />
                </div>
                <div className='col-8'>
                    <h2 className='movie-details__title'>{movie.title}</h2>
                    <h4 className='movie-details__tagline'>{movie.tagline}</h4>
                    <div>
                        <span className='movie-details__date'>
                            {movie.release_date.split('-')[0]}
                        </span>
                        <span className='movie-details__runtime'>
                            {movie.runtime} min
                        </span>
                    </div>
                    <div className='movie-details__overview'>
                        {movie.overview}
                    </div>
                </div>
            </div>
        </div>
    );
};
MovieDetails.propTypes = {
    movie: PropTypes.shape({
        poster_path: PropTypes.string,
        title: PropTypes.string,
        tagline: PropTypes.string,
        release_date: PropTypes.string,
        runtime: PropTypes.number,
        overview: PropTypes.string
    })
};
export default MovieDetails;
