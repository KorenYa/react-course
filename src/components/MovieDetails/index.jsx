import React, { Component } from 'react';
import { connect } from 'react-redux';

import Image from '../Image';
import Button from '../Button';
import SearchResult from '../SearchResult';
import NotFound from '../NotFound';

import { getMovieById } from '../../store/AC';

import './movie-details.scss';

export class MovieDetails extends Component {
  constructor(props) {
    super(props);

    this.handleBackToSearch = this.handleBackToSearch.bind(this);
  }

  componentDidMount() {
    const {
      selectedMovie,
      getMovieById,
      match: {
        params: { filmId: browserfilmId },
      },
    } = this.props;
    if (browserfilmId !== selectedMovie.id) {
      getMovieById(browserfilmId);
    }
  }

  componentDidUpdate(prevProps) {
    const prevFilmId = prevProps.match.params.filmId;
    const {
      match: { params: filmId },
    } = this.props;
    const { getMovieById } = this.props;

    if (prevFilmId !== filmId) {
      getMovieById(filmId);
    }
  }

  handleBackToSearch() {
    const { history, searchQuery } = this.props;
    history.push(`/search/${searchQuery}`);
  }

  render() {
    const { selectedMovie } = this.props;
    const movie = selectedMovie;
    const sameGenreMovies = movie ? movie.sameGenreMovies : null;

    return movie.id ? (
      <React.Fragment>
        <div className="movie-details">
          <div className="row">
            <div className="col-12 text-right">
              <Button
                text="Back to search"
                styleName="btn-lg btn-back"
                onClick={this.handleBackToSearch}
              />
            </div>
            <div className="col-4">
              <Image src={movie.poster_path} alt={movie.title} />
            </div>
            <div className="col-8">
              <h2 className="movie-details__title">
                {movie.title}
              </h2>
              <h4 className="movie-details__tagline">
                {movie.tagline}
              </h4>
              <div>
                <span className="movie-details__date">
                  {movie.release_date.split('-')[0]}
                </span>
                <span className="movie-details__runtime">
                  {movie.runtime}
                  {' '}
min
                </span>
              </div>
              <div className="movie-details__overview">
                {movie.overview}
              </div>
            </div>
          </div>
        </div>
        <div className="status-bar">
          <span>
                        Films by
            {movie.genres[0]}
            {' '}
genre
          </span>
        </div>
        <SearchResult moviesList={sameGenreMovies} />
      </React.Fragment>
    ) : (
      <NotFound />
    );
  }
}

export default connect(
  state => ({
    moviesList: state.moviesList,
    searchQuery: state.searchQuery,
    selectedMovie: state.selectedMovie,
  }),
  {
    getMovieById,
  },
)(MovieDetails);
