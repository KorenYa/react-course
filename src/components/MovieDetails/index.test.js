import React from 'react';
import renderer from 'react-test-renderer';

import { MovieDetails } from '../MovieDetails';

describe('MovieDetails', () => {
    const history = {
        push: jest.fn()
    };
    const match = { params: { filmId: 1 } };
    const mockGetMovieById = jest.fn();

    it('renders correctly', () => {
        const tree = renderer
            .create(
                <MovieDetails
                    selectedMovie={{
                        id: 1,
                        poster_path:
                            'https://image.tmdb.org/t/p/w500/eQRFo1qwRREYwj47Yoe1PisgOle.jpg',
                        title: 'The Gold Rush',
                        tagline: "The World's Greatest Laughing Picture!",
                        release_date: '1925-06-25',
                        runtime: 95,
                        overview:
                            'A lone prospector ventures into Alaska looking for gold. He gets mixed up with some burly characters and falls in love with the beautiful Georgia. He tries to win her heart with his singular charm.',
                        genres: ['adventure'],
                        sameGenreMovies: {
                            data: [],
                            total: 0
                        }
                    }}
                    history={history}
                    match={match}
                    getMovieById={mockGetMovieById}
                />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
