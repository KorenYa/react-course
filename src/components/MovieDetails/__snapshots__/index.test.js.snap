// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly 1`] = `
<div
  className="movie-details"
>
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <img
        alt="The Gold Rush"
        className="img-responsive"
        src="https://image.tmdb.org/t/p/w500/eQRFo1qwRREYwj47Yoe1PisgOle.jpg"
      />
    </div>
    <div
      className="col-8"
    >
      <h2
        className="movie-details__title"
      >
        The Gold Rush
      </h2>
      <h4
        className="movie-details__tagline"
      >
        The World's Greatest Laughing Picture!
      </h4>
      <div>
        <span
          className="movie-details__date"
        >
          1925
        </span>
        <span
          className="movie-details__runtime"
        >
          95
           min
        </span>
      </div>
      <div
        className="movie-details__overview"
      >
        A lone prospector ventures into Alaska looking for gold. He gets mixed up with some burly characters and falls in love with the beautiful Georgia. He tries to win her heart with his singular charm.
      </div>
    </div>
  </div>
</div>
`;
